### PR TITLE
Social monitoring and tweet contacts via campaign

### LINKED ISSUE #1742: Social plugins not working as they should after update
Just installed a new version of Mautic - must admit to being really impressed.

However, if I have a lead, when I click on Social in that lead, I get details of my Twitter account and nothing else. I have the correct @twittter handle as well as other web addresses for facebook, LinkedIn, G+ etc and although the apps are setup they are not working.

I have also cleared the cache out too.

## Steps to reproduce

- I updated to the new version through the link that came through on the Alerts icon with in Mautic

- Having updated initially the plugins link would not work so we cleared the cache this solved the problem accessing the plugins section.

- from that point on any contact that has twitter handle included in their profile started showing the twitter details from the authorized account from the plugin rather than the details on that contacts Twitter activity.

Please help Im losing my mind trying to find out whats happened.

### PR DESCRIPTION
**Pull request by alanhartless**
Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | N
| New feature?  | Y 
| BC breaks?    | N
| Deprecations? | N
| Fixed issues  |  #1742

## Description

This PR does several things:

1. It brings the features of mPower Social to the community courtesy of mautic.com! This provides the ability to create contacts from Twitter mentions or handles and to tweet leads as part of a campaign (your responsibility to be compliant with Twitter's automation rules https://support.twitter.com/articles/76915#Tweets :-) 

## Steps to test this PR

New Features:
1. Apply the PR and clear the cache
2. Run `doctrine:migrations:migrate`
3. Ensure Twitter is authorized in Plugins
4. Go to Channels -> Social Monitoring and create a monitor
5. Run the command `mautic:social:monitor`
6. Click on the monitor in Social Monitoring and look at the leads tab. Click on a lead and you should see the social profile field filled in for Twitter
7. Create a basic campaign with a tweet contact action (**WARNING: be sure you are not tweeting every one you just pulled from a monitor :-D** )
8. Trigger the campaign and the contact should get a tweet

### CODE REVIEW

**Comment by mqueme** [comment]
this worked and its awesome!

**Comment by alanhartless** [comment]
@escopecz Can you give it another shot to see if that last commit fixes it?  I was able to reproduce after testing the form SSO which led me to realizing we were not removing the token from session after it was used and stored in the lead's social details...so it continued to be used for all leads.

File: plugins/MauticSocialBundle/Controller/MonitoringController.php [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+    public function indexAction($page = 1)
+    {
+        $session = $this->get('session');
+
+        if ($this->request->getMethod() == 'POST') {
+            $this->setListFilters();
+        }
+
+        /** @var \MauticPlugin\MauticSocialBundle\Model\MonitoringModel $model */
+        $model = $this->factory->getModel('plugin.mauticSocial.monitoring');
```
Comment (dongilbert) [inline plugins/MauticSocialBundle/Controller/MonitoringController.php]: These all need to be updated to `->getModel('social.monitoring')`

File: plugins/MauticSocialBundle/Views/Monitoring/leads.html.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                                </span>
+                                </div>
+                            </div>
+                            <div class="col-xs-8 va-t">
+                                <div class="panel-body">
+                                    <?php if (in_array($item['id'], $noContactList)) : ?>
+                                        <div class="pull-right label label-danger"><i class="fa fa-ban"> </i></div>
+                                    <?php endif; ?>
+                                    <h4 class="fw-sb mb-xs">
+                                        <a href="<?php echo $view['router']->generate('mautic_lead_action', array("objectAction" => "view", "objectId" => $item['id'])); ?>" data-toggle="ajax">
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Views/Monitoring/leads.html.php]: This is probably where it's failing for me:
```
app/cache/prod/appProdUrlGenerator.php line 205 {"exception":"[object] (Symfony\\Component\\Routing\\Exception\\RouteNotFoundException(code: 0): Unable to generate a URL for the named route \"mautic_lead_action\" as such route does not exist. at /var/www/m.johnlinhart.com/public_html/app/cache/prod/appProdUrlGenerator.php:205)"} []
```
mautic-social

**Comment by mqueme** [comment]
I tried this with the new 2.0 code, it gave me a unserialized error when saving the twitter handle on the contact. When I updated with mine own twitter handle it seemed fine, when updated with yours (sorry to use yours for testing seemed appropriate) it threw an error, it seems to be truncating the data saved in the social_cache array on the DB.

**Comment by alanhartless** [comment]
@mqueme That's likely because it imported an emoji that broke the serialized array. #1730 addresses this.

**Comment by alanhartless** [comment]
@escopecz Fixed the route

**Comment by escopecz** [comment]
Works like a charm now :+1:  :+1:  Really cool feature!

The only thing I can point out is:
mautic-twitter

You can use 
```
<div class="pt-0 pl-15 pb-10 pr-15">
     <?php echo $view->render('MauticCoreBundle:Helper:chart.html.php', array('chartData' => $chartData, 'chartType' => 'line', 'chartHeight' => 300)); ?>
</div>
```
And it will render the chart for you with the styling of the rest of Mautic. The lines should be 1px and the points 2 px.

**Comment by mqueme** [comment]
yes, merging emoji support fixed the error I was getting @alanhartless +1

**Pull request merged**